src/shared/bthdef.rs

```diff
@@ -0,0 +1,1127 @@
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your option.
+// All files in the project carrying such notice may not be copied, modified, or distributed
+// except according to those terms.
+use shared::bthsdpdef::SDP_ERROR;
+use shared::guiddef::GUID;
+use shared::minwindef::{DWORD, UCHAR, ULONG, USHORT};
+use shared::ntdef::{CHAR, ULONGLONG};
+pub const BTH_MAJORVERSION: DWORD = 2;
+pub const BTH_MINORVERSION: DWORD = 1;
+DEFINE_GUID!{GUID_BTHPORT_DEVICE_INTERFACE,
+    0x850302a, 0xb344, 0x4fda, 0x9b, 0xe9, 0x90, 0x57, 0x6b, 0x8d, 0x46, 0xf0}
+DEFINE_GUID!{GUID_BTH_RFCOMM_SERVICE_DEVICE_INTERFACE,
+    0xb142fc3e, 0xfa4e, 0x460b, 0x8a, 0xbc, 0x07, 0x2b, 0x62, 0x8b, 0x3c, 0x70}
+DEFINE_GUID!{GUID_BLUETOOTH_RADIO_IN_RANGE,
+    0xea3b5b82, 0x26ee, 0x450e, 0xb0, 0xd8, 0xd2, 0x6f, 0xe3, 0x0a, 0x38, 0x69}
+DEFINE_GUID!{GUID_BLUETOOTH_RADIO_OUT_OF_RANGE,
+    0xe28867c9, 0xc2aa, 0x4ced, 0xb9, 0x69, 0x45, 0x70, 0x86, 0x60, 0x37, 0xc4}
+DEFINE_GUID!{GUID_BLUETOOTH_L2CAP_EVENT,
+    0x7eae4030, 0xb709, 0x4aa8, 0xac, 0x55, 0xe9, 0x53, 0x82, 0x9c, 0x9d, 0xaa}
+DEFINE_GUID!{GUID_BLUETOOTH_HCI_EVENT,
+    0xfc240062, 0x1541, 0x49be, 0xb4, 0x63, 0x84, 0xc4, 0xdc, 0xd7, 0xbf, 0x7f}
+DEFINE_GUID!{GUID_BLUETOOTH_AUTHENTICATION_REQUEST,
+    0x5DC9136D, 0x996C, 0x46DB, 0x84, 0xF5, 0x32, 0xC0, 0xA3, 0xF4, 0x73, 0x52}
+DEFINE_GUID!{GUID_BLUETOOTH_KEYPRESS_EVENT,
+    0xD668DFCD, 0x0F4E, 0x4EFC, 0xBF, 0xE0, 0x39, 0x2E, 0xEE, 0xC5, 0x10, 0x9C}
+DEFINE_GUID!{GUID_BLUETOOTH_HCI_VENDOR_EVENT,
+    0x547247e6, 0x45bb, 0x4c33, 0xaf, 0x8c, 0xc0, 0x0e, 0xfe, 0x15, 0xa7, 0x1d}
+DEFINE_GUID!{Bluetooth_Base_UUID,
+    0x00000000, 0x0000, 0x1000, 0x80, 0x00, 0x00, 0x80, 0x5F, 0x9B, 0x34, 0xFB}
+pub const SDP_PROTOCOL_UUID16: USHORT = 0x0001;
+pub const UDP_PROTOCOL_UUID16: USHORT = 0x0002;
+pub const RFCOMM_PROTOCOL_UUID16: USHORT = 0x0003;
+pub const TCP_PROTOCOL_UUID16: USHORT = 0x0004;
+pub const TCSBIN_PROTOCOL_UUID16: USHORT = 0x0005;
+pub const TCSAT_PROTOCOL_UUID16: USHORT = 0x0006;
+pub const ATT_PROTOCOL_UUID16: USHORT = 0x0007;
+pub const OBEX_PROTOCOL_UUID16: USHORT = 0x0008;
+pub const IP_PROTOCOL_UUID16: USHORT = 0x0009;
+pub const FTP_PROTOCOL_UUID16: USHORT = 0x000A;
+pub const HTTP_PROTOCOL_UUID16: USHORT = 0x000C;
+pub const WSP_PROTOCOL_UUID16: USHORT = 0x000E;
+pub const BNEP_PROTOCOL_UUID16: USHORT = 0x000F;
+pub const UPNP_PROTOCOL_UUID16: USHORT = 0x0010;
+pub const HID_PROTOCOL_UUID16: USHORT = 0x0011;
+pub const HCCC_PROTOCOL_UUID16: USHORT = 0x0012;
+pub const HCDC_PROTOCOL_UUID16: USHORT = 0x0014;
+pub const HCN_PROTOCOL_UUID16: USHORT = 0x0016;
+pub const AVCTP_PROTOCOL_UUID16: USHORT = 0x0017;
+pub const AVDTP_PROTOCOL_UUID16: USHORT = 0x0019;
+pub const CMPT_PROTOCOL_UUID16: USHORT = 0x001B;
+pub const UDI_C_PLANE_PROTOCOL_UUID16: USHORT = 0x001D;
+pub const L2CAP_PROTOCOL_UUID16: USHORT = 0x0100;
+DEFINE_BLUETOOTH_UUID128!{SDP_PROTOCOL_UUID,
+    SDP_PROTOCOL_UUID16}
```

Comment:
All these `DEFINE_BLUETOOTH_UUID128!` invocations should be single line invocations.